packages/opentelemetry-plugin-mongodb/test/mongodb.ts

```diff
@@ -0,0 +1,273 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Span } from '@opentelemetry/node-tracer';
+import * as assert from 'assert';
+import * as mongodb from 'mongodb';
+import { plugin } from '../src/';
+import { ProxyTracer } from './utils/ProxyTracer';
+import { SpanKind } from '@opentelemetry/types';
+
+export interface MongoDBAccess {
+  client: mongodb.MongoClient;
+  collection: mongodb.Collection;
+}
+
+/**
+ * Access the mongodb collection.
+ * @param url The mongodb URL to access.
+ * @param dbName The mongodb database name.
+ * @param collectionName The mongodb collection name.
+ */
+function accessCollection(
+  url: string,
+  dbName: string,
+  collectionName: string
+): Promise<MongoDBAccess> {
+  return new Promise((resolve, reject) => {
+    mongodb.MongoClient.connect(url, function connectedClient(err, client) {
+      if (err) {
+        reject(err);
+        return;
+      }
+      const db = client.db(dbName);
+      const collection = db.collection(collectionName);
+      resolve({ client, collection });
+    });
+  });
+}
+
+/**
+ * Asserts root spans attributes.
+ * @param rootSpanVerifier An instance of rootSpanVerifier to analyse RootSpan
+ * instances from.
+ * @param expectedName The expected name of the first root span.
+ * @param expectedKind The expected kind of the first root span.
+ */
+function assertSpan(
+  spans: Span[],
+  expectedName: string,
+  expectedKind: SpanKind
+) {
+  assert.strictEqual(spans.length, 2);
+  spans.forEach(span => {
+    assert.deepStrictEqual(typeof span.endTime, 'number');
+  });
+  assert.strictEqual(spans[1].name, expectedName);
+  assert.strictEqual(spans[1].kind, expectedKind);
+}
+
+describe('MongoDBPlugin', () => {
+  // For these tests, mongo must be running. Add OPENTELEMETRY_MONGODB_TESTS to run
+  // these tests.
+  const OPENTELEMETRY_MONGODB_TESTS = process.env
+    .OPENTELEMETRY_MONGODB_TESTS as string;
+  let shouldTest = true;
+  if (!OPENTELEMETRY_MONGODB_TESTS) {
+    console.log('Skipping test-mongodb. Run MongoDB to test');
+    shouldTest = false;
+  }
+
+  const URL = 'mongodb://localhost:27017';
+  const DB_NAME = 'opentelemetry-tests';
+  const COLLECTION_NAME = 'test';
+
+  const tracer = new ProxyTracer({});
+  let client: mongodb.MongoClient;
+  let collection: mongodb.Collection;
+
+  before(done => {
+    plugin.enable(mongodb, tracer);
+    accessCollection(URL, DB_NAME, COLLECTION_NAME)
+      .then(result => {
+        client = result.client;
+        collection = result.collection;
+        done();
+      })
+      .catch((err: Error) => {
+        console.log(
+          'Skipping test-mongodb. Could not connect. Run MongoDB to test'
+        );
+        shouldTest = false;
```

Comment:
I think that we should not skip tests when we are running in ci, WDYT ?
https://circleci.com/docs/2.0/env-vars/#built-in-environment-variables
so `process.env.CI` should detect it.

Reply:
Indeed, i just didn't had the time to correctly setup the circleci config to add the defined env vars in to all builds.

Reply:
I believe that it’s automatically injected by circleci by default

Reply:
I meant the `OPENTELEMETRY_MONGODB_TESTS` variable, not the CI one. You might want to enable it in local and enable a `CI` variable in this case would be weird